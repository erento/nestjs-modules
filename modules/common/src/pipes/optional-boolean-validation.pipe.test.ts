import {OptionalBooleanValidationPipe} from './optional-boolean-validation.pipe';

describe('OptionalBooleanValidationPipe', (): void => {
    let pipe: OptionalBooleanValidationPipe;

    beforeEach((): void => {
        pipe = new OptionalBooleanValidationPipe();
    });

    test.each<[string | undefined, boolean | undefined]>([
        ['true', true],
        ['false', false],
        [undefined, undefined],
    ])(
        'should transform string to value properly',
        (input: string | undefined, expected: boolean | undefined): void => {
            expect(pipe.transform(input)).toEqual(expected);
        },
    );
    test('should fail on empty string', (): void => {
        expect((): boolean | undefined => pipe.transform('')).toThrowErrorMatchingSnapshot();
    });
    test('should fail on invalid string', (): void => {
        expect((): boolean | undefined => pipe.transform('asdf')).toThrowErrorMatchingSnapshot();
    });
});
