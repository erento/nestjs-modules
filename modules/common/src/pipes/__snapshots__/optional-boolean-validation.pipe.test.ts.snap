// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionalBooleanValidationPipe should fail on empty string 1`] = `"value is not a boolean: ''"`;

exports[`OptionalBooleanValidationPipe should fail on invalid string 1`] = `"value is not a boolean: 'asdf'"`;
